Toolkit/ArcGISToolkit/AR/ArcGISARView.swift

```diff
@@ -0,0 +1,682 @@
+    private var initialTransformationMatrix = AGSTransformationMatrix()
+    
+    /// Whether ARKit supported on this device?
+    private var isSupported = false
+    
+    /// Whether the client has been notfiied of start/failure
+    private var notifiedStartOrFailure = false
+    
+    /// These are used when calculating framerate
+    var frameCount:Int = 0
+    var frameCountTimer: Timer?
+    
+    // A quaternion used to compensate for the pitch beeing 90 degrees on ARKit; used to calculate the current device transformation for each frame
+    let compensationQuat:simd_quatf = simd_quaternion(Float(sin(45 / (180 / Float.pi))), 0, 0, Float(cos(45 / (180 / Float.pi))))
+    
+    /// The quaternion used to represent the device orientation; used to calculate the current device transformation for each frame; defaults to landcape-left
+    var orientationQuat:simd_quatf = simd_quaternion(0, 0, 1, 0)
+    
+    // MARK: Initializers
+    
+    public override init(frame: CGRect) {
+        super.init(frame: frame)
+        sharedInitialization()
+    }
+    
+    required public init?(coder aDecoder: NSCoder) {
+        super.init(coder: aDecoder)
+        sharedInitialization()
+    }
+    
+    /// Initializer used to denote whether to display the live camera image
+    ///
+    /// - Parameter renderVideoFeed: whether to dispaly the live camera image
+    public convenience init(renderVideoFeed: Bool){
+        self.init(frame: CGRect.zero)
+        self.renderVideoFeed = renderVideoFeed
+    }
+    
+    /// Initialization code shared between all initializers
+    private func sharedInitialization(){
+        //
+        // ARKit initialization
+        isSupported = ARWorldTrackingConfiguration.isSupported
+        
+        addSubviewWithConstraints(arSCNView)
+        arSCNView.session.delegate = self
+        (arSCNView as SCNSceneRenderer).delegate = self
+        
+        //
+        // add sceneView to view and setup constraints
+        addSubviewWithConstraints(sceneView)
+
+        //
+        // make our sceneView's background transparent, no atmosphereEffect
+        sceneView.isBackgroundTransparent = true
+        sceneView.atmosphereEffect = .none
+        
+        //
+        // tell the sceneView we will be calling `renderFrame()` manually
+        sceneView.isManualRendering = true
+        
+        //
+        // we haven't yet notified user of start or failure
+        notifiedStartOrFailure = false
+        
+        //
+        // set intitial orientationQuat
+        orientationChanged(notification: nil)
+    }
+    
+    // MARK: Public
+    
+    /// Determines the map point for the given screen point
+    ///
+    /// - Parameter screenPoint: the point in screen coordinates
+    /// - Returns: the map point corresponding to screenPoint
+    public func arScreenToLocation(screenPoint: AGSPoint) -> AGSPoint {
+        return AGSPoint(x: 0.0, y: 0.0, spatialReference: nil)
```

Comment:
The working of this is designed to correspond to the `AGSMapView`'s `-(AGSPoint*)screenToLocation:(CGPoint)screen` method.

Reply:
> does this make more sense?

I don't think so since `toLocation` is a parameter label, but the parameter is the `from`, not the `to`. AppKit had a similar API design challenge with methods like `UIWindow`'s  `- (NSRect)convertRectToScreen:(NSRect)rect`. If I remember correctly, the Swift interface generator originally turned that into `convert(toScreen:)`, but they changed it to `convertToScreen(_:)` since the parameter label didn't make sense.

Reply:
Ok, thanks.  Back to `arScreenToLocation`